ArduPlane/GCS_Mavlink.cpp

```diff
@@ -721,7 +721,12 @@ MAV_RESULT GCS_MAVLINK_Plane::handle_command_int_do_reposition(const mavlink_com
             requested_position.relative_alt = 0;
         }
 
-        plane.set_guided_WP(requested_position);
+        float radius = -1.0f; // default radius to -1 for no-change
+        // Loiter radius for planes. Positive values only, direction is controlled by Yaw (param4) value. A value of zero or NaN is ignored.
+        if (!isnan(packet.param3) && packet.param3 > 0) {
```

Comment:
You don't need to check for a positive value of radius because you will pass a negative anyways, and any negative is considered invalid.

Reply:
The comment here is also that 0 is ignored, however the actual handler eats 0, and you can't release it back to the default radius.

Reply:
ignoring 0 makes it backwards compatible to packets that don't know about param3. The handler eats zero internally to change it to the param based radius but param3 in this mavlink packet is intentionally skipped

Reply:
Okay, well I guess this raises my question/concern though. There is no way to go back to the tuned value once you change the radius for the first time short of leaving and reentering the flight mode. I'd like to allow something to be a reset. Using 0 returns us to the old behavior so I don't think that's breaking. If we don't want to use 0, we should grab a negative number to be the reset command.

Reply:
fair enough. Perhaps -1 could mean default and 0 means no-change?

Reply:
I'm fine with that. I think 0 could mean go back to the default, but if it's easier then we can do that and I'm fine with it.

Reply:
GCS handler changed to:
```
        float radius = 0;
        // Loiter radius for planes. Positive radius in meters, direction is controlled by Yaw (param4) value.
        // radius values:-1 = use default param WP_LOITER_RAD, 0 = ignore (no-change), >0 radius
        if (!isnan(packet.param3) && !is_zero(packet.param3)) {
            radius = packet.param3;
        }
        plane.set_guided_WP(requested_position, radius);